.github/workflows/test-mobile.yml

```diff
@@ -15,31 +15,115 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  codecheck:
-    name: "Ledger Live Mobile CodeCheck"
-    env:
-      NODE_OPTIONS: "--max-old-space-size=7168"
-      FORCE_COLOR: 3
-
-    runs-on: ubuntu-latest
+  detox-tests-ios:
+    name: "Ledger Live Mobile - iOS Detox Tests"
+    runs-on: macos-latest
     steps:
       - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.event.push.head.sha }}
       - uses: pnpm/action-setup@v2.1.0
         with:
           version: latest
       - uses: actions/setup-node@v3
         with:
-          node-version: 16
+          node-version: 16.x
           cache: pnpm
           cache-dependency-path: "**/pnpm-lock.yaml"
+      - name: Install applesimutils
+        run: |
+          brew tap wix/brew
+          brew install applesimutils
+      - name: bump npm
+        run: npm i -g npm@8.12.2
```

Comment:
Why this specific version @nabil-brn ?

Reply:
I just copied everything from the build workflow, given it has been done by @LedgerHQ/live-devx , I assume it's what we need.
And targetting specific versions gives us more control on our workflow 👍.

Reply:
Wating for comfirmation, we will test again if this fix is needed